README.md

```diff
@@ -39,3 +39,57 @@ specification and construction, the
 [Container image creation tool `concreate`](https://github.com/cekit/cekit) is
 used as the primary point of investigation, see the `image.yaml` file for
 more information.
+
+# Partial images without an Apache Spark distribution installed
+
+This repository also supports building 'incomplete' versions of
+the images which contain tooling for OpenShift but lack an actual
+Spark distribution. An s2i workflow can be used with these partial
+images to install a Spark distribution of a user's choosing.
+This gives users an alternative to checking out the repository
+and modifying build files if they want to run a custom
+Spark distribution. By default, the partial images built will be
+
+* `openshift-spark-inc`, Apache Spark, Python 2.7
+* `openshift-spark-inc-py36`, Apache Spark, Python 3.6
+
+## Build
+
+To build the partial images, use make with Makefile.inc
+
+    make -f Makefile.inc
+
+## Push
+
+Tag and push the images to the designated reference.
+
+    make -f Makefile.inc push SPARK_IMAGE=[REGISTRY_HOST[:REGISTRY_PORT]/]NAME[:TAG]
+
+## Image Completion
+
+To produce a final image, a source-to-image build must be performed which takes
+a Spark distribution as input. This can be done in OpenShift or locally using
+the s2i tool if it's installed. The final images created can be used just
```

Comment:
i think it would be nice to make the "s2i tool" text a link to the upstream repo, just for convenience.

https://github.com/openshift/source-to-image

Reply:
great idea, will do